components/fxa-client/src/fxa_client.udl

```diff
@@ -310,11 +352,27 @@ interface FirefoxAccount {
   //
   [Throws=FxaError]
   string get_current_device_id();
-  
+
+  // Get the list of known devices registered on the user's account.
+  //
+  // The application might use this information to e.g. display a list of appropriate
+  // send-tab targets.
+  //
+  // # Notes
+  //
+  //    - Device metadata is only visible to applications that have been
+  //      granted the `https://identity.mozilla.com/apps/oldsync` scope.
+  [Throws=FxaError]
+  DeviceList get_device_list();
```

Comment:
I decided to add this method directly to `FirefoxAccount`.  The wrapper code for both Kotlin and Swift create a new class for this.  I don't have strong feelings either way, but I think I slightly prefer having the entire API in one class.